### PR TITLE
A11y

### PR DESCRIPTION
**Pull request by talensjr**
Added A11y sample. Accept HTMLDivElement props on SnapList and SnapItem. Accept ref on SnapItem. Remove event.preventDefault on mouseDown event from SnapList

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -74,6 +74,63 @@ export const App = () => (
...
+      <SnapItem margin={{ left: '15px', right: '15px' }} snapAlign="center">
+        <MyItem>Item 2</MyItem>
+      </SnapItem>
+      <SnapItem margin={{ left: '15px', right: '15px' }} snapAlign="center">
+        <MyItem>Item 3</MyItem>
+      </SnapItem>
+      <SnapItem
+        margin={{ left: '15px', right: '15px' }}
+        snapAlign="center"
+        ref={lastSnapItem}
```
Comment (luispuig) [inline README.md]: If it does the same, maybe better to do the focus to MyItem.

File: README.md [hunk 2]
```diff
@@ -176,6 +233,7 @@ export const App = () => {
   bottom?: string;
   left?: string;
   } | undefined }: The margin is used to set the separation between the items. You can use different margin for the first and last item to get better results.
+- `ref` { React.RefObject\<HTMLDivElement\> | undefined }. You can use it to programatically focus the `SnapItem` for example.
```
Comment (luispuig) [inline README.md]: If the dev can use the ref to his element inside the children we can avoid this.

File: README.md [hunk 3]
```diff
@@ -339,6 +397,13 @@ This an internal util function used by `useDragToScroll` that can be useful for
 - **Breakchange** useDragToScroll is returning `{ isDragging }` instead of the boolean. https://github.com/luispuig/react-snaplist-carousel/commit/a2d23d6b804dcab1da9520db8edc746c6837f23e#diff-e3457effa5fa347d185fdd0d08ba3209R173
 - Version `4.1.0`. Fix support for macOS Big Sur
 - Version `4.2.0`. Add useScroll / goTo / animationEnabled option. Usefull to scroll on component mount.
+- Version `4.3.0`.
+  - Removed `event.preventDefault()` form `mouseDown` so focus event can be propagated up in the DOM tree.
+  - `<SnapList>` and `<SnapItem>` now accept all the `HTMLDivElement` properties.
+  - `<SnapItem>` now accepts a `ref` property.
```
Comment (luispuig) [inline README.md]: Same as above

File: example/src/A11y/A11y.jsx [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+  >
+    {children}
+  </div>
+);
+
+const itemWidth = 100;
+
+const List = ({ id }) => {
+  const snapList = useRef(null);
+  const snapItem14 = useRef(null);
```
Comment (luispuig) [inline example/src/A11y/A11y.jsx:34]: Same as above

File: example/src/A11y/A11y.jsx [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+import styles from './styles.module.css';
+
+const Item = ({ onClick, children, visible, ...props }) => (
+  <div
+    className={styles.item}
+    style={{
+      background: visible ? '#bce6fe' : '#cccccc',
+      cursor: visible ? 'default' : 'pointer',
+    }}
+    onClick={onClick}
```
Comment (luispuig) [inline example/src/A11y/A11y.jsx:15]: To be consistent with the A11y, shall we make this component respond to a keypress enter or space?

**Review by luispuig — CHANGES_REQUESTED** [review]
Thanks a lot for the contribution!!! Good work!!!

File: README.md (hunk 1 above)
Comment (talensjr) [reply README.md]: Updated!!

File: README.md (hunk 2 above)
Comment (talensjr) [reply README.md]: Updated!

File: README.md (hunk 3 above)
Comment (talensjr) [reply README.md]: Updated!

File: example/src/A11y/A11y.jsx (hunk 4 above)
Comment (talensjr) [reply example/src/A11y/A11y.jsx:34]: Updated!

File: example/src/A11y/A11y.jsx (hunk 5 above)
Comment (talensjr) [reply example/src/A11y/A11y.jsx:15]: Updated!

**Review by luispuig — APPROVED** [review]

**Pull request merged**